suzieq/poller/worker/nodes/node.py

```diff
@@ -232,6 +241,32 @@ def is_connected(self):
         '''Is there connectivity to the device at the transport level'''
         return self._conn is not None
 
+    @asynccontextmanager
+    async def limit_pipeline(self):
+        '''Context Manager to implement throttling of commands.
+
+        In many networks, backend authentication servers such as TACACS which
+        handle authentication of logins and even command execution, cannot
+        large volumes of authentication requests. Thanks to our use of
+        asyncio, we can easily sends hundreds of connection requests to such
+        servers, which effectively turns into authentication failures. To
+        handle this, we add a user-specified maximum of simultaneous
+        commands/logins at any given time. This code implements that locking
+        context.
+        '''
+        if self._cmd_sem:
+            self.logger.info(
+                f'{self.transport}://{self.hostname}:{self.port}: Get lock')
+            await self._cmd_sem.acquire()
+            self.logger.info(
+                f'{self.transport}://{self.hostname}:{self.port}: Got lock')
+            yield
+            self.logger.info(
+                f'{self.transport}://{self.hostname}:{self.port}: Free lock')
+            self._cmd_sem.release()
```

Comment:
```suggestion
            try:
              yield
            finally:
              self.logger.info(
                  f'{self.transport}://{self.hostname}:{self.port}: Free lock')
              self._cmd_sem.release()
```
Using finally we release the semaphore even when an exception is raised inside the context